pymeasure/instruments/hp/hp4145x.py

```diff
@@ -0,0 +1,1252 @@
+                },
+
+                "VAR2": {
+                    "start" : 0,
+                    "step" : 10e-6,
+                    "points" : 3,
+                    "compliance" : 2
+
+                }
+            }
+
+    """
+
+    def __init__(self, adapter, name="HP 4145A/4145B Semiconductor Parameter Analyzer",
+                 **kwargs):
+        kwargs.setdefault('timeout', 100000)
+        super().__init__(
+            adapter,
+            name,
+            **kwargs
+        )
+
+        self._mode = None
+        self.smu1 = SMU(self, 1)
+        self.smu2 = SMU(self, 2)
+        self.smu3 = SMU(self, 3)
+        self.smu4 = SMU(self, 4)
+        self.vmu1 = VMU(self, 1)
+        self.vmu2 = VMU(self, 2)
+        self.vsu1 = VSU(self, 1)
+        self.vsu2 = VSU(self, 2)
+        self.var1 = VAR1(self)
+        self.var2 = VAR2(self)
+        self.vard = VARD(self)
+
+        self._manual_flush = False
+
+    @property
+    def manual_flush(self):
+        """
+        Control the manual flushing of channels 'channel definition'
+        """
+        return self._manual_flush
+
+    @manual_flush.setter
+    def manual_flush(self, value):
+        self.smu1.manual_flush = value
+        self.smu2.manual_flush = value
+        self.smu3.manual_flush = value
+        self.smu4.manual_flush = value
+        self.vmu1.manual_flush = value
+        self.vmu2.manual_flush = value
+        self.vsu1.manual_flush = value
+        self.vsu2.manual_flush = value
+        self.var1.manual_flush = value
+        self.var2.manual_flush = value
+        self._manual_flush = value
+
+    def flush_channel_definition(self):
+        """
+        Flush the channel definitions of all sub channels.
+        Only required with :attr:`manual_flush` True.
+        """
+        self.smu1.flush_channel_definition()
+        self.smu2.flush_channel_definition()
+        self.smu3.flush_channel_definition()
+        self.smu4.flush_channel_definition()
+        self.vmu1.flush_channel_definition()
+        self.vmu2.flush_channel_definition()
+        self.vsu1.flush_channel_definition()
+        self.vsu2.flush_channel_definition()
+
+    def flush_source_setup(self):
+        """
+        Flush the channel definitions of all sub channels.
+        Only required with :attr:`manual_flush` True.
+        """
+        self.var1.flush_source_setup()
+        self.var2.flush_source_setup()
+        self.smu1.flush_source_setup()
+        self.smu2.flush_source_setup()
+        self.smu3.flush_source_setup()
+        self.smu4.flush_source_setup()
+        self.vsu1.flush_source_setup()
+        self.vsu2.flush_source_setup()
+
+    integration_time = Instrument.setting(
+        "SS %s",
+        """Set the integration time.
+
+        Values: :code:`SHORT` for no averages, :code:`MEDIUM` for 16 averages and :code:`LONG`
+        for 256 averages
+
+        .. code-block:: python
+
+            instr.integration_time = "MEDIUM"
```

Comment:
Did you render the documentation locally (with `make html` in the docs folder) and look into it (in built folder)?